name: k6 Load Tests
on:
  schedule:
  # Mon-Fri 5AM Finnish time (UTC+3)
  - cron: '0 2 * * 1-5'
  push:
    branches:
      - TH-1051-k6-collection-load-tests
jobs:
  k6_load_test:
    name: k6 Load Tests
    runs-on: ubuntu-latest

    steps:
      - name: Checkout
        uses: actions/checkout@v2
      - name: Get yarn cache directory path
        id: yarn-cache-dir-path
        run: echo "::set-output name=dir::$(yarn config get cacheFolder)"
      - uses: actions/cache@v2
        id: yarn-cache
        with:
          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
          restore-keys: |
            ${{ runner.os }}-yarn-
      - name: Install dependencies
        run: yarn --prefer-offline --frozen-lockfile --check-files --only-dev
      - name: Build
        run: yarn build:load-test
      - name: Run k6 tests
        uses: k6io/action@v0.1
        with:
          filename: dist/run-all.k6.test.js
          flags: -e BASE_URL=https://${{ secrets.ENVIRONMENT_URL_STAGING }} -e GRAPHQL_BASE_URL=https://tapahtumat-proxy.test.kuva.hel.ninja/proxy/graphql
      - name: k6 Load Tests failed notifications
        uses: rtCamp/action-slack-notify@v2
        env:
          SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK }}
          SLACK_USERNAME: K6LoadTest
          SLACK_ICON: https://a.slack-edge.com/80588/img/services/outgoing-webhook_48.png
          SLACK_TITLE: K6 Load Tests has failed.
          SLACK_MESSAGE: Some of the k6 load tests has failed.
        if: failure()